Automattic/jetpack
# My Jetpack: Integrate Boost score increase into Boost card and tooltip

This PR shows whether the Boost Score (in the My Jetpack Boost product card) has increased (with an up-arrow and the number of points increased), and if so, also displays different copy in the tooltip.

**Screenshots:**

![Markup 2024-02-29 at 10 22 54](https://github.com/Automattic/jetpack/assets/11078128/1c3ee19b-7651-409f-bda2-d0d2b0324f90)

.
![Markup 2024-02-29 at 10 25 52](https://github.com/Automattic/jetpack/assets/11078128/80ab5ea3-41dd-4a80-85e5-682bdfdb8a12)





Fixes: https://github.com/Automattic/jetpack-roadmap/issues/1236

## Proposed changes:


* When the Boost Plugin is activated, we fetch the Boost speed score and cache it in localStorage for 1 day.  That way new Boost scores are only fetched once a day, at most.
* When the Boost plugin is not activated, we only fetch new boost scores after 2 weeks.
* Display an up-arrow and the number of points increased, if the currently fetched (daily) score has increased compared to the previous 'latestBoostSpeedScores'.
* If score increase, also display different copy in the tooltip.

### Other information:


## Jetpack product discussion



Project Thread: My Jetpack: Add Boost speed score into boost product card: pbNhbs-9Nt-p2

## Does this pull request change what data or activity we track or use?


No

## Testing instructions:






* Create a Jurassic.Ninja site with Beta plugin running this feature branch (`add/my-jetpack-boost-score-increase`).
* Navigate to the My Jetpack page. Verify the Boost Score is not loading (no spinner), and not showing because there is no site connection.
* Connect Jetpack (Either site-only or user connection, doesn't matter).
* Go to My Jetpack page, Verify you see the loading spinner in the Boost card. (Boost scores are loading for the first time).
* Verify The Boost scores eventually load and the score bar and letter grade are shown. (Take note of the score, you may need to know it later.)
* Now purchase the Boost Paid plan ($9.95 plan).
* Now change the Date/time on your computer to 1 day forward.
* Reload the My Jetpack page.
* Verify the Boost scores are loading. (You should see the loading spinner).
* Once loaded: If the new score is lower than the first score you had previously (from the previous day): preform the next step: Otherwise skip the next step:
* For the localStorage key `MyJetpackTodaysBoostScore`, change the value of `score` to a number that's a few points higher than the first score you got. Then reload the My Jetpack page.
* In the Boost product card, you should now see there is an increase in score, denoted by an up-arrow and the number of points it has improved. ( &uarr;5 ) - (See screenshots above)
* Hover your mouse over (or tab to focus) the score letter grade. Verify the copy starts with, "Your speed has improved by x points!"  ( you can check what copy should show based on the various states, here: 33a29-pb )
* Don't forget to change the date/time on your computer back 1 day to where it was before. 😉
